pkg/store/bucket.go

```diff
@@ -630,6 +632,34 @@ func populateChunk(out *storepb.AggrChunk, in chunkenc.Chunk, aggrs []storepb.Ag
 	return nil
 }
 
+func debugFoundBlockSetOverview(logger log.Logger, mint, maxt int64, lset labels.Labels, bs []*bucketBlock) {
+	if len(bs) == 0 {
+		level.Debug(logger).Log("msg", "No block found", "mint", mint, "maxt", maxt, "lset", lset.String())
+		return
+	}
+
+	var parts []string
+
+	currRes := bs[0].meta.Thanos.Downsample.Resolution
+	currMin := bs[0].meta.MinTime
+	currMax := bs[0].meta.MaxTime
+	for i := 1; i < len(bs); i++ {
```

Comment:
I'm not sure how is this ordered, but can you iterate from the other side (`for i := len(bs)-1; ...; i--`)?

Reply:
not sure how iterating backward will change anything here? I simplified algo a little bit.